samples/net/mqtt_azure/README.rst

```diff
@@ -0,0 +1,87 @@
+.. _mqtt-azure-sample:
+
+MQTT Azure Sample
+#################
+
+Overview
+********
+
+This sample application demonstrates how an MQTT client
+can publish messages to an Azure Cloud IoT hub based on MQTT protocol.
+
+- Acquire a DHCPv4 lease
+- Establish a TLS connection with Azure Cloud IoT hub
+- Publish data to the Azure cloud
+- SOCKS5 supported
+
+The source code of this sample application can be found at:
+:zephyr_file:`samples/net/mqtt-azure`.
+
+Requirements
+************
+
+- Azure Cloud account
+- Azure IOT Cloud credentials and required information
+- Freedom Board (FRDM-K64F)
+- Network connectivity
+
+Building and Running
+********************
+This application has been built and tested on the NXP FRDMK64F.
+Certs are required to authenticate to the Azure Cloud IoT hub.
+Current certs in `samples/net/mqtt_azure/src/test_certs.h` are
```

Comment:
In rst files, single backticks are meaningful as markers for references to other things.  Sphinx throws an "any" error here because it can't find something to link to (such as a function name, or document label).  If you just want the text to appear in a ``mono-spaced font``, use double backticks.  In this particular case, you can identify and add a link to the file in the Zephyr github repo by using:  `` :zephyr_file:`samples/net/mqtt_azure/src/test_certs.h` `` (as done on line 18 above).